### ISSUE TITLE
regarding the return error code  that tee_fs_generate_fek function returns when buf_size is not equal to TEE_FS_KM_FEK_SIZE

### ISSUE THREAD

**Issue by seungeun2** [OP-TEE/optee_os#553]
HI, this is probably minor issue but i just wanna clarify..

Within "tee_fs_generate_fek" function,(core/tee/tee_fs_key_manager.c, line#:288)

```c
if (buf_size != TEE_FS_KM_FEK_SIZE)
    return TEE_ERROR_SHORT_BUFFER;
```

shouldn't it return TEE_ERROR_BAD_PARAMETERS instead of TEE_ERROR_SHORT_BUFFER?
(TEE_ERROR_BAD_PARAMETERS makes more sense and it is also consistent with  "fek_crypt" funciton, core/tee/tee_fs_key_manager.c, line#: 82 

```c
if (size != TEE_FS_KM_FEK_SIZE)
    return TEE_ERROR_BAD_PARAMETERS;
```

)

**Comment by jbech-linaro** [comment]
Hi @seungeun2 ,

Your suggestions sounds fine with me, feel free to make a patch for this.

**Comment by jbech-linaro** [comment]
@jforissier , this is in the code you're dealing with. Would you mind check whether this still is applicable and if yes, would you mind adding a fix for it?

**Comment by jforissier** [comment]
Indeed TEE_ERROR_BAD_PARAMETERS would make more sense here. Actually the function is only ever called with buf_size == TEE_FS_KM_FEK_SIZE, so we could as well drop the size parameter. I will update the return code.

**Comment by jforissier** [comment]
https://github.com/OP-TEE/optee_os/pull/885

**Comment by jbech-linaro** [comment]
Thanks for the quick reply and the fix provided.

**Comment by jbech-optee-adm** [comment]
Closing the PR since a fix has been merged (#885).
